### PR TITLE
feat(tr): insurance payments client

### PR DESCRIPTION
**Pull request by thorkellmani**
## What

* Update the current `social-insurance-administration client`
* Create a new api domain for social insurance
* New service portal module for social insurance

## Why

People want their social insurance overview

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

**Comment by andes-it** [comment]
This PR's had a 15-day siesta. Lost its spark, did it? It's making an exit in 5 days unless the stale label disappears.

**Review by lodmfjord — APPROVED** [review]
LGTM

**Review by Toti91 — APPROVED** [review]
code owned LGTM

File: libs/api/domains/social-insurance/src/lib/dtos/paymentPlan.input.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { InputType, Field, Int } from '@nestjs/graphql'
+
+@InputType()
+export class SocialInsurancePaymentPlanInput {
```
Comment (saevarma) [inline libs/api/domains/social-insurance/src/lib/dtos/paymentPlan.input.ts:4]: Inconsistent with the models so I would opt for the short class name and set the gql type name as argument
```suggestion
@InputType('SocialInsurancePaymentPlanInput')
export class PaymentPlanInput {
```

File: libs/api/domains/social-insurance/src/lib/socialInsurance.service.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    @Inject(LOGGER_PROVIDER) private readonly logger: Logger,
+    private readonly socialInsuranceApi: SocialInsuranceAdministrationClientService,
+  ) {}
+
+  async getPaymentPlan(
+    user: User,
+    year?: number,
+  ): Promise<PaymentPlan | undefined> {
+    const res = await this.socialInsuranceApi
+      .getPaymentPlan(user, year ?? 2023)
```
Comment (saevarma) [inline libs/api/domains/social-insurance/src/lib/socialInsurance.service.ts:23]: default to `2023` if nothing is passed? Is there any logic behind that? Would it be better to get current year - 1?

File: libs/api/domains/social-insurance/src/lib/socialInsurance.service.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                  }
+                })
+                .filter(isDefined) ?? [],
+          }
+        })
+        .filter(isDefined) ?? []
+
+    return {
+      nextPayment: res.nextPayment ?? 1234,
+      previousPayment: res.previousPayment ?? 9876,
```
Comment (saevarma) [inline libs/api/domains/social-insurance/src/lib/socialInsurance.service.ts:87]: The default values? Do we want to use those numbers?

File: libs/api/domains/social-insurance/src/lib/util.ts
Comment (saevarma) [inline libs/api/domains/social-insurance/src/lib/util.ts:1]: empty file

File: libs/api/domains/social-insurance/src/lib/socialInsurance.service.ts (hunk 3 above)
Comment (thorkellmani) [reply libs/api/domains/social-insurance/src/lib/socialInsurance.service.ts:87]: sorry, that's mock stuff i forgot to remove

File: libs/service-portal/core/src/lib/messages.ts [hunk 4]
```diff
@@ -1254,4 +1262,81 @@ export const m = defineMessages({
...
+      'Einstaklingar geta átt rétt á greiðslum frá Tryggingastofnun (TR) til að bæta afkomu sína og hag. Allar greiðslur frá TR þarf að sækja sérstaklega um.',
+  },
+  paymentPlan: {
+    id: 'service-portal:payment-plan',
+    defaultMessage: 'Greiðsluáætlun',
+  },
+  paymentsReasoning: {
+    id: 'service-portal:payments-reasoning',
+    defaultMessage: 'Forsenda greiðslna',
+  },
```
Comment (thordurhhh) [inline libs/service-portal/core/src/lib/messages.ts:1281]:
```suggestion
    id: 'service.portal:social-insurance-maintenance',
    defaultMessage: 'Framfærsla',
  },
  socialInsuranceMaintenanceDescription: {
    id: 'service.portal:social-insurance-maintenance-description',
    defaultMessage:
      'Einstaklingar geta átt rétt á greiðslum frá Tryggingastofnun (TR) til að bæta afkomu sína og hag. Allar greiðslur frá TR þarf að sækja sérstaklega um.',
  },
  paymentPlan: {
    id: 'service.portal:payment-plan',
    defaultMessage: 'Greiðsluáætlun',
  },
  paymentsReasoning: {
    id: 'service.portal:payments-reasoning',
    defaultMessage: 'Forsenda greiðslna',
  },
```

File: libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/PaymentPlan.tsx [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+    return years.map((y) => ({ value: y, label: y.toString() }))
+  }, [])
+
+  const [selectedYear, setSelectedYear] = useState(yearOptions[0])
+  /*
+  useEffect(() => {
+    getPaymentPlanQuery({
+      variables: { input: { year: selectedYear.value } },
+    })
+  }, [selectedYear, getPaymentPlanQuery])*/
```
Comment (thordurhhh) [inline libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/PaymentPlan.tsx:50]: Remove?

File: libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/PaymentPlan.tsx [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+import { PaymentGroupTableRow } from '../../components'
+import { PaymentPlanWrapper } from './wrapper/wrapper/PaymentPlanWrapper'
+import { SocialInsuranceMaintenancePaths } from '../../lib/paths'
+
+const PaymentPlan = () => {
+  useNamespaces('sp.social-insurance-maintenance')
+  const { formatMessage } = useLocale()
+
+  /*const [getPaymentPlanQuery, { data, loading, error }] =
+    useGetPaymentPlanLazyQuery()*/
```
Comment (thordurhhh) [inline libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/PaymentPlan.tsx:37]: Remove?

File: libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/PaymentPlan.tsx [hunk 7]
```diff
@@ -0,0 +1,293 @@
...
+  }, [selectedYear, getPaymentPlanQuery])*/
+
+  const error = undefined
+  const loading = false
+
+  if (error) {
+    return <Problem type="internal_service_error" error={error} />
+  }
+
+  const data: GetPaymentPlanQuery = {
```
Comment (thordurhhh) [inline libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/PaymentPlan.tsx:59]: What data is this?

File: libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/wrapper/wrapper/PaymentPlanWrapper.tsx [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+          )?.children ?? []
+        }
+      />
+      {!loading && !error && children && <Box marginTop={[6]}>{children}</Box>}
+      <Box>
+        <Text variant="small" marginTop={5} marginBottom={2}>
+          Sorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc
+          vulputate libero et velit interdum, ac aliquet odio mattis. Class
+          aptent taciti sociosqu ad litora torquent per conubia nostra, per
+          inceptos himenaeos.
```
Comment (thordurhhh) [inline libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/wrapper/wrapper/PaymentPlanWrapper.tsx:56]: Lorem ipsum?

File: libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/wrapper/wrapper/PaymentPlanWrapper.tsx [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+  }
+  return (
+    <Box>
+      <IntroHeader
+        title={formatMessage(coreMessages.socialInsuranceMaintenance)}
+        intro={formatMessage(
+          coreMessages.socialInsuranceMaintenanceDescription,
+        )}
+        serviceProviderSlug={'tryggingastofnun'}
+        serviceProviderTooltip="Eitthvað tooltip"
```
Comment (thordurhhh) [inline libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/wrapper/wrapper/PaymentPlanWrapper.tsx:39]: Eitthvað tooltip? 😵‍💫

File: libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/PaymentPlan.tsx [hunk 10]
```diff
@@ -0,0 +1,293 @@
...
+                {formatMessage(coreMessages.print)}
+              </Button>
+            </GridColumn>
+            <GridColumn>
+              <Button
+                size="small"
+                variant="utility"
+                icon="download"
+                iconType="outline"
+                onClick={() => console.log('clicked download')}
```
Comment (thordurhhh) [inline libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/PaymentPlan.tsx:211]: Then download?

**Review by thordurhhh — CHANGES_REQUESTED** [review]
Some changes needed 🚀

File: libs/api/domains/social-insurance/src/lib/dtos/paymentPlan.input.ts (hunk 1 above)
Comment (thorkellmani) [reply libs/api/domains/social-insurance/src/lib/dtos/paymentPlan.input.ts:4]: will do

File: libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/PaymentPlan.tsx (hunk 10 above)
Comment (thorkellmani) [reply libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/PaymentPlan.tsx:211]: removed

File: libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/wrapper/wrapper/PaymentPlanWrapper.tsx (hunk 9 above)
Comment (thorkellmani) [reply libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/wrapper/wrapper/PaymentPlanWrapper.tsx:39]: fixed

File: libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/wrapper/wrapper/PaymentPlanWrapper.tsx (hunk 8 above)
Comment (thorkellmani) [reply libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/wrapper/wrapper/PaymentPlanWrapper.tsx:56]: removed

File: libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/PaymentPlan.tsx (hunk 7 above)
Comment (thorkellmani) [reply libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/PaymentPlan.tsx:59]: forgot to remove mock data

File: libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/PaymentPlan.tsx (hunk 6 above)
Comment (thorkellmani) [reply libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/PaymentPlan.tsx:37]: forgot to remove mock data, this is uncommented

File: libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/PaymentPlan.tsx (hunk 5 above)
Comment (thorkellmani) [reply libs/service-portal/social-insurance-maintenance/src/screens/PaymentPlan/PaymentPlan.tsx:50]: forgot to remove mock data, this is uncommented

**Review by karenbjorg — APPROVED** [review]

**Review by saevarma — APPROVED** [review]
Core files LGTM

**Review by oddsson — APPROVED** [review]

**Review by thordurhhh — APPROVED** [review]
🚂

**Pull request merged**